python-lib/dku_param_loading.py

```diff
@@ -32,24 +34,33 @@ def load_input_output_params(recipe_id: RecipeID) -> Dict:
         PluginParamValidationError: If a parameter is not valid
 
     """
-    # Input dataset
     params = {}
-    input_dataset_names = get_input_names_for_role("input_dataset")
-    if len(input_dataset_names) == 0:
-        raise PluginParamValidationError("Please specify input folder")
-    params["input_dataset"] = dataiku.Dataset(input_dataset_names[0])
-    input_dataset_columns = [p["name"] for p in params["input_dataset"].read_schema()]
     # Index folder
     if recipe_id == RecipeID.SIMILARITY_SEARCH_INDEX:
         output_folder_names = get_output_names_for_role("index_folder")
         if len(output_folder_names) == 0:
             raise PluginParamValidationError("Please specify index folder as output")
         params["index_folder"] = dataiku.Folder(output_folder_names[0])
+        params["folder_partition_root"] = get_folder_partition_root(params["index_folder"])
     elif recipe_id == RecipeID.SIMILARITY_SEARCH_QUERY:
         input_folder_names = get_input_names_for_role("index_folder")
         if len(input_folder_names) == 0:
             raise PluginParamValidationError("Please specify index folder as input")
         params["index_folder"] = dataiku.Folder(input_folder_names[0])
+        params["folder_partition_root"] = get_folder_partition_root(params["index_folder"], is_input=True)
+        check_only_one_read_partition(params["folder_partition_root"], params["index_folder"])
```

Comment:
What was the purpose of the `check_only_one_read_partition` ?

Reply:
It is to verify that the object has one and only read partition. Otherwise, it is impossible to map the partition to the right partitioned index.